### PR TITLE
branch name validation pr

### LINKED ISSUE #87: Uppercase stage names break cognito with custom domain name
This isn't a problem yet, but has been seen when implementing Okta which uses a cognito custom domain.  Can/should be fixed ahead of time.  found by @sethsacher 

`  An error occurred: UserPoolDomain - The domain name contains an invalid character. Domain names can only contain lower-case letters, numbers, and hyphens. Please enter a different name that follows this format: ^[a-z0-9](?:[a-z0-9\-]{0,61}[a-z0-9])?$ (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: 013e0b42-cac6-4420-ab34-27720db86024; Proxy: null).`

I recommend adding to the branch verification in github actions and circle ci to fail if theres an uppercase.

AC:
- The error above is avoided.

### PR DESCRIPTION
**Pull request by eolaniyan052**
## Purpose

Pull request to address the the bug causing Uppercase stage names to break cognito with custom domain name

#### Linked Issues to Close

closes #87 
https://github.com/CMSgov/macpro-quickstart-serverless/issues/87

## Approach

Basically my approach was to add a step to the deploy.yml that will run a script to verify branch name. That will run first so if it fails there, then a message gets displayed and the other jobs wont get ran.
#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: branchVal.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+set -e
+
+local_branch=${1}
+
+valid_branch_dash="([a-z0-9\-\]+)"
+
+
+message="There is something wrong with your branch name. Branch names in this project must adhere to this contract: $valid_branch_regex, $local_branch. Your commit will be rejected. You should rename your branch to a valid name and try again."
+
+if [[ ! $local_branch =~ $valid_branch_regex ]]
```
Comment (mdial89f) [inline branchVal.sh:12]: The if statement should also check total length....
See the second part of the if here:  https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/.github/workflows/deploy.yml#L23

File: branchVal.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+local_branch=${1}
+
+valid_branch_dash="([a-z0-9\-\]+)"
```
Comment (mdial89f) [inline branchVal.sh:7]: This regex statement should keep all logic from the existing regex, while removing the capital letter allowance.
See https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/.github/workflows/deploy.yml#L23 for the existing statement.  I think the existing statement says "Must start with a letter (either upper case or lower case), then may contain any number or sequence of upper/lower case letters, numbers, or hyphens"

You want to keep all that, but remove the upper case piece

File: branchVal.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+message="There is something wrong with your branch name. Branch names in this project must adhere to this contract: $valid_branch_regex, $local_branch. Your commit will be rejected. You should rename your branch to a valid name and try again."
```
Comment (mdial89f) [inline branchVal.sh:10]: I'm very biased here, since I wrote the original message.  But I prefer the older style of the error message.
See:  https://github.com/CMSgov/macpro-quickstart-serverless/blob/master/.github/workflows/deploy.yml#L25-L40
I like that it spells out that this is a serverless constraint, and also tries to explain in plain english the constraints.
It would have to be modified to explain that upper cases letters are no good.
Consider going back to that more verbose message format.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -42,6 +42,8 @@ jobs:
             exit 1
           fi
       - uses: actions/checkout@v1
+      - name: Validate branch name
+        run: ./branchVal.sh $STAGE_PREFIX$branch_name
```
Comment (mdial89f) [inline .github/workflows/deploy.yml:46]: Consider changing the name of branchVal.sh to something that doesn't contain "val".  I figure you mean val as in "validate"... no problems, just "val" is sort of a protected term around here, and usually denotes an environment.
validateBranchName.sh
branchNameValidation.sh
checkBranchNameIsValid.sh

Something like that.

Comment (mdial89f) [reply .github/workflows/deploy.yml:46]: Also... calling this script effectively replaces the "Check branch name is a legal serverless stage name" step in deploy.yml   Be sure to remove that; I don't see it removed here.

**Review by mdial89f — CHANGES_REQUESTED** [review]

File: branchVal.sh (hunk 3 above)
Comment (dwhitecl) [reply branchVal.sh:10]: agreed with this ^^

**Review by mdial89f — APPROVED** [review]
